Xamarin.Forms.Xaml.UnitTests/DataTemplateExtension.xaml.cs

```diff
@@ -0,0 +1,33 @@
+﻿using NUnit.Framework;
+
+using Xamarin.Forms.Core.UnitTests;
+
+namespace Xamarin.Forms.Xaml.UnitTests
+{
+	public partial class DataTemplateExtension : ContentPage
+	{
+		public DataTemplateExtension() => InitializeComponent();
+		public DataTemplateExtension(bool useCompiledXaml)
+		{
+			//this stub will be replaced at compile time
```

Comment:
I am not sure I understand this comment... Does this compilation actually modify the type?

Reply:
OH YEAH

We have a special task that takes this assembly and replace the content of this ctor with a conditional that either goes to InitializeComponent (which was itself replaced by the XamlC task) or a call to inflate the xaml dynamically at runtime.

Again, this is boilerplate, and you can ignore what it does or the way it does it, as it's impossible to have a mental model of all the abstractions going on with the Xaml Compiler...